cmp/grid/Grid.ts

```diff
@@ -472,65 +474,53 @@ export class GridLocalModel extends HoistModel {
 
                 const agColState = api.getColumnState();
 
-                // 0) Insert the auto group col state if it exists, since we won't have it in our column state list
+                // Insert the auto group col state if it exists, since we won't have it in our column state list
                 const autoColState = agColState.find(c => c.colId === 'ag-Grid-AutoColumn');
                 if (autoColState) {
-                    colState.splice(agColState.indexOf(autoColState), 0, autoColState);
+                    colState.splice(
+                        agColState.indexOf(autoColState),
+                        0,
+                        autoColState as ColumnState
```

Comment:
Does this case risk spoiling the change detection below due to unfortunate hide/hidden mismatch?

I know it's an edge-case where we have this auto-column inserted by the grid - it's due to direct use of AG-grid grouping APIs, is that right?  So maybe nothing we need to worry about, just checking if there's some tweak we would make to get it 100%

Reply:
Good catch, only an issue when using ag-grid grouping without full-row groups, but worthwhile to be correct